File: sdk/python/examples/wrapped.py
"""Print a user's year-in-review summary.

Run:
    uv run python examples/wrapped.py alice.bsky.social 2024
"""

from __future__ import annotations

import asyncio
import json
import sys

from rocksky import Client


async def main(handle_or_did: str, year: int | None) -> int:
    async with Client() as rocksky:
        profile = await rocksky.actor.get_profile(handle_or_did)
        wrapped = await rocksky.stats.wrapped(profile.did, year=year)

    if not wrapped:
        print(f"no wrapped payload for {handle_or_did}")
        return 1

    print(json.dumps(wrapped, indent=2, default=str))
    return 0


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("usage: wrapped.py <handle-or-did> [year]")
        sys.exit(2)
    handle = sys.argv[1]
    year = int(sys.argv[2]) if len(sys.argv) > 2 else None
    sys.exit(asyncio.run(main(handle, year)))

File: sdk/rust/examples/wrapped.rs
//! Pretty-print someone's Wrapped year-in-review payload.
//!
//! ```text
//! cargo run --example wrapped -- did:plc:abc... [2025]
//! ```

use rocksky::Client;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut args = std::env::args().skip(1);
    let Some(did) = args.next() else {
        eprintln!("usage: wrapped <did-or-handle> [year]");
        std::process::exit(1);
    };
    let year = args.next().and_then(|s| s.parse::<i32>().ok());

    let client = Client::new();
    let wrapped = client.stats().wrapped(&did, year).await?;

    println!("{}", serde_json::to_string_pretty(&wrapped)?);
    Ok(())
}
